lib/vast/Conversion/Common/IRsToLLVM.cpp

```diff
@@ -48,6 +49,54 @@ namespace vast::conv::irstollvm
         erase_pattern< hl::TypeDeclOp >
     >;
 
+    template< typename self_t >
+    struct value_builder
+    {
+      private:
+        const self_t &self() const { return static_cast< const self_t & >(*this); }
+
+      protected:
+
+        mlir::Value construct_value(auto &rewriter, mlir_value val) const {
+            auto op = val.getDefiningOp();
+            VAST_ASSERT(op && op->getResults().size() == 1);
+            auto trg_type = self().convert(op->getResults()[0].getType());
+            return construct_value(rewriter, op, trg_type);
+        }
+
+        mlir::Value construct_value(auto &rewriter, operation op, mlir_type type) const {
+            auto init_list = mlir::dyn_cast< hl::InitListExpr >(op);
+            if (!init_list)
+            {
+                VAST_CHECK(op->getResults().size() == 1, "Unexpected number of results");
+                return op->getResults()[0];
+            }
+
+            auto trg_type = self().convert(type);
+            if (mlir::isa< mlir::LLVM::LLVMArrayType, mlir::LLVM::LLVMStructType >(trg_type))
+                return construct_aggregate_value(rewriter, init_list, trg_type);
+            VAST_FATAL("Not implemented yet.");
+        }
+
+        mlir::Value construct_aggregate_value(
+            conversion_rewriter &rewriter, hl::InitListExpr init_list,
+            mlir_type aggragate_type
+        ) const {
+            // Currently we are only supporting this type of initialization so
+            // better be defensive about it.
+            mlir_value init = self().undef(rewriter, init_list.getLoc(), aggragate_type);
+            for (auto [idx, o] : llvm::enumerate(init_list.getElements()))
```

Comment:
what is `o` give it more meaningful abbreviation?

Reply:
`operand`, I think. Will do.